pkg/kubelet/kubelet.go

```diff
@@ -144,6 +144,9 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
+	// Max amount of time to admit a single Pod, mostly accounting for device plugin allocation.
+	maxSinglePodAdmissionTimeout = 30 * time.Second
```

Comment:
As far as I understand *all* pods will be blocked for 30 sec if device plugin is not registered/stuck or processing `Allocate` calls too slow. Should this timeout be shorter and may be configurable?

Reply:
By `all pods` I mean even those that are not requesting device resources. See https://github.com/kubernetes/kubernetes/issues/129249 for details.

Reply:
Mainly referenced their previous discussions. https://github.com/kubernetes/kubernetes/pull/127717#discussion_r1779453855

Actually, I think the same as you. It would be best if `maxSinglePodAdmissionTimeout` were a parameter of kubelet.

- `maxSinglePodAdmissionTimeout = -1` means no timeout limit, maintaining the previous behavior, equivalent to a feature gate.

- `maxSinglePodAdmissionTimeout > 0` sets the timeout duration.

What are your thoughts? @SergeyKanzhelev @ffromani

Reply:
> By all pods I mean even those that are not requesting device resources. See https://github.com/kubernetes/kubernetes/issues/129249 for details.

This is concerning for me because admission process is meant to be fast and cheap and introducing admission timeout can potentially impact _all pods_ including the ones not requesting devices does not seem ideal. If the timeout period was just for pods requesting devices I would probably be okay with the approach but when there are is potential of slowing down admission process of all pods we have to be careful. 

Perhaps @smarterclayton has some thoughts here.